odoo/models.py

```diff
@@ -2483,6 +2483,12 @@ def _read_group_format_result(self, rows_dict, lazy_groupby):
                                 value, format=READ_GROUP_DISPLAY_FORMAT[granularity],
                                 locale=locale
                             )
+                        # babel week numbering doesn't factor in user configured first weekday
+                        # e.g. if set to Friday, "'W'w YYYY" will return "W1 2026" for both
+                        #      2026-01-02 and 2027-01-01
```

Comment:
This seems to be happening because of a change in babel.

As said in babel (https://babel.pocoo.org/en/latest/dates.html#date-fields) or odoo documentation:
https://github.com/odoo/odoo/blob/e616795b083936ae529c70fdd3a5dccbda5eda30/odoo/models.py#L369-L380

 `w` should be used with `YYYY` and this case (getting the wrong year) should not happen.

Checking different babel versions:

```py
> (__import__('babel').__version__, __import__('babel.dates').dates.format_datetime(__import__('datetime').datetime(2027, 1, 1, 0, 0), "'W'w YYYY"))
('2.6.0', 'W1 2027')
> (__import__('babel').__version__, __import__('babel.dates').dates.format_datetime(__import__('datetime').datetime(2027, 1, 1, 0, 0), "'W'w YYYY"))
('2.7.0', 'W1 2026')
```

Reply:
side note: this discussion (by xmo from odoo) seems to be about this issue, he probably has a good idea of the state of the issue https://github.com/python-babel/babel/discussions/885

Reply:
Also, the documentation says `"w YYYY"` formats to ISO week-year, but that's evidently not the case:

```python
In [28]: __import__('babel.dates').dates.format_datetime(__import__('datetime').date(2027, 1, 31), "'W'w YYYY", locale='fr_FR')
Out[28]: 'W4 2027'

In [29]: __import__('babel.dates').dates.format_datetime(__import__('datetime').date(2027, 1, 31), "'W'w YYYY", locale='fr')
Out[29]: 'W5 2027'

In [30]: __import__('babel.dates').dates.format_datetime(__import__('datetime').date(2027, 1, 31), "'W'w YYYY", locale='en_US')
Out[30]: 'W6 2027'
```
No idea why `fr` and `fr_FR` would even give different results.

Reply:
> Different locale configurations (different firstday / mindays) would handily do it.

Of course, I'm more so confused as to why `fr` would be configured differently from `fr_BE` or `fr_FR`.